early-attack-detector-py/src/agent.py

```diff
@@ -359,6 +383,47 @@ def handle_transaction(
 ):
     return real_handle_transaction(transaction_event)
 
+def provide_handle_block(w3):
+    def handle_block(block_event: forta_agent.block_event.BlockEvent) -> list:
+        findings = []
+
+        # Choosing the denominator based on chain's block time
+        # so as to update TP list approx. once daily
+        DAILY_BLOCKS_DENOMINATOR = ETH_BLOCKS_IN_ONE_DAY if CHAIN_ID == 1 else THREE_SECOND_BLOCKS_IN_ONE_DAY
+        if block_event.block_number % DAILY_BLOCKS_DENOMINATOR == 0:
+            global TP_ATTACKER_LIST
+            TP_ATTACKER_LIST = update_tp_attacker_list(TP_ATTACKER_LIST)
+        return findings
+        
+    return handle_block
+
+
+real_handle_block = provide_handle_block(web3)
+
+
+def handle_block(block_event: forta_agent.block_event.BlockEvent):
+    return real_handle_block(block_event)
+
+def provide_handle_alert(w3):
+    def handle_alert(alert_event: forta_agent.alert_event.AlertEvent) -> list:
+        findings = []
+
+        global TP_ATTACKER_LIST
+        if alert_event.alert.metadata.sender not in TP_ATTACKER_LIST:
+            TP_ATTACKER_LIST.append(alert_event.alert.metadata.sender.lower())
+        
+        TP_ATTACKER_LIST.append(alert_event.alert.metadata.receiver.lower())
```

Comment:
Key access is required here, and also added `.lower()`just in case
```suggestion
        if alert_event.alert.metadata['sender'].lower() not in TP_ATTACKER_LIST:
            TP_ATTACKER_LIST.append(alert_event.alert.metadata['sender'].lower())
        
        TP_ATTACKER_LIST.append(alert_event.alert.metadata['receiver'].lower())
```